docs/source/javascript_api.rst

```diff
@@ -0,0 +1,87 @@
+Javascript API
+==============
+
+Dallinger provides a javascript API to facilitate creating web-based
+experiments. ll of the dallinger demos use this API to communicate
+with the experiment server. The API is defined in the `dallinger2.js`
+script, which is included in the default experiment templates.
+
+The `dallinger` object
+----------------------
+
+Any page that includes `dallinger2.js` script will have a `dallinger`
+object added to the `window` global namespace. This object defines a
+number of functions for interacting with Dallinger experiments.
+
+Making requests to experiment routes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`dallinger` provides functions which can be used to asynchronously
+interact with any of the experiment routes described in
+:doc:`Web API <web_api>`:
+
+.. js:autofunction:: dallinger.get
+
+.. js:autofunction:: dallinger.post
+
+
+Additionally, the `dallinger` object provides functions that make requests
+to specific experiment routes:
+
+.. js:autofunction:: dallinger.createAgent
+
+.. js:autofunction:: dallinger.createInfo
+
+.. js:autofunction:: dallinger.getInfo
+
+.. js:autofunction:: dallinger.getInfos
+
+.. js:autofunction:: dallinger.getReceivedInfos
+
+.. js:autofunction:: dallinger.getTransmissions
+
+
+.. _deferreds-label:
+
+`Deferred` objects
+~~~~~~~~~~~~~~~~~~
+
+All of the above functions make use of `jQuery.Deferred <https://api.jquery.com/jquery.deferred/>`__,
+and return `Deferred` objects. These `Deferred` objects provide the following
+methods to facilitate handling asynchronous responses once they've completed:
+
+    * ``.done(callback)``: Provide a callback to handle data from a successful response
+    * ``.fail(callback)``: Provide a callback to handle error responses
```

Comment:
Our error-handling scheme is now:
1. Pass `myCallback(rejection)` to `fail()`
2. `myCallback()`  will be called with an `AjaxRejection` object, which can be inspected, in case you want to do something other than show the error page
3. If you do want to show the error page, you pass this rejection on to `dallinger.error(rejection)`

This awkward to document, and is maybe overkill (consistent use in the demos might be enough... if we have that? 😟 ), but maybe it should be documented here.

Reply:
I definitely think it's worth documenting this since I didn't really fully follow it when looking at the updated code. I'll see what I can do.